providers/gitlab_test.go

```diff
@@ -153,46 +153,6 @@ var _ = Describe("Gitlab Provider Tests", func() {
 		})
 	})
 
-	Context("when filtering on email", func() {
```

Comment:
I think the test case is still needed if GitLabProvider is doing verified Email checks with the response from the Userinfo endpoint. You just need to remove the `p.EmailDomains` part